godtools/App/Features/Menu/Domain/UseCases/AuthenticateUserUseCase/AuthenticateUserUseCase.swift

```diff
@@ -30,6 +32,8 @@ class AuthenticateUserUseCase {
         return authenticateByAuthTypePublisher(authType: authType)
             .flatMap({ (accessToken: OktaAccessToken) -> AnyPublisher<CruOktaUserDataModel, Error> in
                 
+                self.getMobileContentAuthToken.refreshAuthToken(with: accessToken.value)
```

Comment:
Hey @rachaelblue also note here the fetching of the authToken we want part of the ```authenticateByAuthTypePublisher``` chain ```(add additional flatMap)```.  This way if ```authenticateByAuthTypePublisher``` is cancelled, everything is cancelled in this chain.

Reply:
Hey @rachaelblue hmm that's true, we don't need the auth token value here and at this point we really don't need to fetch the auth token at all.

We could just leave it out and allow the auth token to be fetched when an auth request is made.  The benefit of having it here is we would have a auth token in place for that future initial request.  Were you thinking along these lines too?

If we do include the fetch request in the chain here, we would still get the return value, we just wouldn't do anything with it which might be confusing.  It would look something like this...

```swift
return authenticateByAuthTypePublisher(authType: authType)
            .flatMap({ (accessToken: OktaAccessToken) -> AnyPublisher<String?, URLResponseError> in
                
                return self.getMobileContentAuthToken.fetchTokenFromRemote(oktaAccessToken: accessToken.value)

                    .eraseToAnyPublisher()
            })
            .flatMap({ (authToken: String?) -> AnyPublisher<CruOktaUserDataModel, Error> in
                                                                
                return self.cruOktaAuthentication.getAuthUserPublisher()
                    .mapError { oktaError in
                        return oktaError.getError()
                    }
                    .eraseToAnyPublisher()
            })
            .flatMap({ (authUser: CruOktaUserDataModel) -> AnyPublisher<Bool, Error> in
                
                self.postEmailSignUp(authUser: authUser)
                self.setAnalyticsUserProperties(authUser: authUser)
                
                return Just(true).setFailureType(to: Error.self)
                    .eraseToAnyPublisher()
            })
            .eraseToAnyPublisher()
```

Reply:
Ah I see, I was thinking the Okta token was used elsewhere in the chain, so commandeering it for `fetchTokenFromRemote()` wasn't going to work.  But since that's not the case, I think it's ok to ignore the auth token value as long as it makes sense to wait until the auth token responds to trigger `getAuthUserPublisher()`.

We probably still want to fetch the token here rather than waiting til we need it.  I think Android is doing it this way too, attempts to use an existing token and only refreshes it if the auth request fails.

So if that all makes sense, I think we can go with your solution ^

Reply:
cc @frett

Reply:
Hey @rachaelblue I'd say let's fetch that auth token here that way it's stored and ready for our authorized requests.

It would be nice if we could execute the ```fetchTokenFromRemote()``` and ```getAuthUserPublisher``` concurrently as well.  Maybe we can use CombineLatest or MergeMany.  I need to look those up again and refresh myself on the differences.

Reply:
on Android we don't pro-actively fetch the auth token, we fetch it on the first call to an API that requires authentication. after we fetch it for the authenticated API request we cache it to reuse as long as we can so that we aren't constantly requesting it every time we use an authenticated API.

I also have synchronization in place, so if multiple authenticated API requests fire at the same time they will allow block until a single auth token is retrieved, and then all of them resume. This is a nice optimization, but isn't absolutely necessary.

Reply:
Also of note is the fact that we only use authentication on user APIs currently. There is no need to change how tools are synced

Reply:
Thanks @frett.  @rachaelblue after Daniel's comment lets keep it consistent with Android and only fetch the auth token when it's actually needed via an authorized request.

Reply:
Sounds good, fixed!